### PR TITLE
feat: deployment script

### PR DESCRIPTION
**Pull request by 0xJabberwock**
# 🤖 Linear

Closes GRT-138

### CODE REVIEW

File: script/Deploy.s.sol [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    _responseBondSize = 0;
+    _responseDeadline = 0;
+    _responseDisputeWindow = 0;
+
+    // TODO: Define dispute module params
+    _disputeBondSize = 0;
+    _maxNumberOfEscalations = 0;
+    _disputeDeadline = 0;
+    _tyingBuffer = 0;
+    _disputeDisputeWindow = 0;
```
[the remark this answers is not in the document]
Comment (ashitakah) [reply script/Deploy.s.sol:94]: I think that can be different each field, but also, put the same values could be easiest for the usage

File: test/unit/Deploy.t.sol [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  function test_DeployRun(uint256 _currentEpoch) public happyPath(_currentEpoch) {
+    deploy.setUp();
+    deploy.run();
+
+    // Oracle
+    assertEq(address(deploy.oracle()).code, type(Oracle).runtimeCode);
+
+    // Modules
+    // BUG: Error (9274): "runtimeCode" is not available for contracts containing immutable variables.
+    // assertEq(address(deploy.eboRequestModule()).code, type(EBORequestModule).runtimeCode);
```
Comment (0xJabberwock) [inline test/unit/Deploy.t.sol:49]: Asserting the `runtimeCode` of a contract type with the `code` deployed at the expected address effectively checks that the deployed bytecode is the desired one. However, this cannot be done in presence of contracts containing `immutable` variables. 😕

File: script/Deploy.s.sol [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    _disputeParams.tyingBuffer = _tyingBuffer;
+    _disputeParams.disputeWindow = _disputeDisputeWindow;
+    requestData.disputeModuleData = abi.encode(_disputeParams);
+
+    // Set resolution module data
+    _resolutionParams.arbitrator = address(councilArbitrator);
+    requestData.resolutionModuleData = abi.encode(_resolutionParams);
+
+    // TODO: Set finality module data?
+    // requestData.finalityModuleData = abi.encode(_finalityParams);
```
Comment (0xJabberwock) [inline script/Deploy.s.sol:179]: I want to confirm that `finalityModuleData` will remain empty for EBO's use cases.

Comment (0xShaito) [reply script/Deploy.s.sol:190]: Yes! No data.

File: script/Deploy.s.sol [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    _setRequestData(_precomputedEBORequestCreator);
+    eboRequestCreator = new EBORequestCreator(oracle, epochManager, arbitrator, council, requestData);
+
+    // Assert that `EBORequestCreator` was deployed at the precomputed address
+    if (eboRequestCreator != _precomputedEBORequestCreator) revert Deploy_InvalidPrecomputedAddress();
+
+    vm.stopBroadcast();
+  }
+
+  function _setRequestData(IEBORequestCreator _precomputedEBORequestCreator) internal {
```
Comment (0xShaito) [inline script/Deploy.s.sol:148]: Let's do this as a return of the object and not a set to a variable in storage maybe

**Review by 0xShaito — CHANGES_REQUESTED** [review]

**Comment by 0xJabberwock** [comment]
When inheriting from `BondEscalationModule`, I'm not able to compile due to the cursed import statement that comes with it:
```solidity
// @defi-wonderland/prophet-modules/solidity/contracts/modules/dispute/BondEscalationModule.sol
import 'import {FixedPointMathLib} from 'node_modules/solmate/srcutils/FixedPointMathLib.sol';
```
Strange thing is that the line isn't bugged in `prophet-modules` repo, so most probably the `solidity-exporter-action` is misbehaving.

File: script/Deploy.s.sol [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+  function setUp() public {
+    // Define The Graph accounts
+    graphToken = IERC20(_GRAPH_TOKEN);
+    epochManager = IEpochManager(_EPOCH_MANAGER);
+    arbitrator = _ARBITRATOR;
+    council = _COUNCIL;
+
+    // TODO: Set production request module params
+    paymentAmount = 0.1 ether;
```
Comment (0xShaito) [inline script/Deploy.s.sol:81]: Lets have paymentAmount be 0 as we don't have that ready yet

**Review by 0xShaito — CHANGES_REQUESTED** [review]

**Review by 0xShaito — APPROVED** [review]

**Review by ashitakah — APPROVED** [review]
Really well done !

**Pull request merged**